### DISCUSSION TITLE
Unable to add report if install path is different

### DISCUSSION THREAD

**Discussion by haki99** [netbox-community/netbox#12478] [Q&A] [answered]
### NetBox version

v3.5.0

### Python version

3.10

### Steps to Reproduce

- Install netbox in a different path (in my case its installed in a k8s environment on alpine, the path is /app/netbox/...)
- Copy files to a folder (in my case its /config/reports and /config/scripts, these folders are linked to /app/netbox/netbox/reports and scripts, I believe these are the default paths)
- Add local data source (in my case: file:///app/netbox/netbox/scripts and file:///app/netbox/netbox/reports)
- Add a new report from the data source

### Expected Behavior

The report is added.

### Observed Behavior

Shows this error, about file not found and a path, which is used nowhere in the configuration files:

![image](https://user-images.githubusercontent.com/51157531/236193714-3d745cd5-80d8-4702-b4dc-7ac00c9fa98d.png)

In my case with the same steps but with scripts is working.

**Comment by jeremystretch** [comment]
I'm afraid you haven't provided sufficient information regarding your installation, so I've converted this to a discussion. Maybe someone more familiar with your deployment can assist.

I will suggest checking the value of the `REPORTS_ROOT` config variable, because the `/opt/netbox/` path is not hard-coded anywhere in the application.

**Reply by haki99** [reply]
Thank you for the suggestion, I was curious about the hardcoded part.

Previously I checked multiple times the configuration and I think that it is correct:

![image](https://user-images.githubusercontent.com/51157531/236403626-4d60b393-964f-4b59-b4f1-65978e1282a5.png)

**Comment by candlerb** [comment]
It's very unclear what you're doing.

You mention k8s, so are you using a docker container image? Is it [the official netbox-docker](https://github.com/netbox-community/netbox-docker), or something you found somewhere else, or something you have built yourself? The official netbox-docker image uses `/opt/netbox`, not `/app/netbox`.

You've also not said what you mean by "Add local data source" and "Add a new report from the data source".  What is this "data source" setting you are referring to, and what exactly did you change?

Netbox supports [remote storage](https://docs.netbox.dev/en/stable/installation/3-netbox/#remote-file-storage) for uploaded files (i.e. images), but for local storage you would not use remote storage at all, you'd just use paths (without any `file:///` URLs). And as far as I know, Netbox can't use remote storage for reports or custom scripts anyway.

**Reply by haki99** [reply]
I use a k8s environment and I deploy a forked and self-built [linuxserver/netbox](https://github.com/linuxserver/docker-netbox) image. I am aware about the difference of the install path, I was curious about if it is hard-coded somewhere in the code.

Add local data source is a new feature in Netbox v3.5.0, officially in the documentation you can [find it here](https://docs.netbox.dev/en/stable/models/core/datasource/). It has an option to add a local datasource (folder on the server), and as I mentioned before, I added the folders like this:

> Add local data source (in my case: file:///app/netbox/netbox/scripts and file:///app/netbox/netbox/reports)

The official documentation suggests you to do like this, on the page I linked in my previous sentence:

![image](https://user-images.githubusercontent.com/51157531/236404875-e2d2f875-e0ae-4f2f-8d77-822dab3cbf7c.png)

Because I use a different, self-built image, the problem might not be in the official image, especially if it is not hard-coded anywhere, so I keep looking for it in my code, maybe I left a "skeleton" somewhere.

If I find anything I will write the solution here.

**Reply by candlerb** [reply]
> I deploy a forked and self-built [linuxserver/netbox](https://github.com/linuxserver/docker-netbox) image

OK, thanks. Probably best to ask your question at that project then.

As Jeremy said, the location of reports is controlled by its own setting (REPORTS_ROOT) and is unrelated to the new 3.5.0 data sources feature, so that was just confusing the issue.  You could start by trying a simple standalone report which doesn't use data sources, e.g. one from https://github.com/netbox-community/customizations/tree/master/reports

**Reply by haki99** [reply, marked as answer]
Yeah, I started with that and showed the same error.

Anyway.

I found the solution, the problem was in my CI/CD pipeline, the configuration doesn't changed in the deployment.
